tests/conftest.py

```diff
@@ -255,18 +255,15 @@ def data(tmpdir_factory, _cookiecutter_config_file, docker_compose):
     template = os.path.join(
         os.path.abspath(os.path.dirname(__file__)), os.pardir)
     output_factory = tmpdir_factory.mktemp
+    defaults = "partition=normal output=logs/slurm-%j.out error=logs/slurm-%j.err"
     c = Cookies(template, output_factory, _cookiecutter_config_file)
     c._new_output_dir = lambda: str(p.join("slurm"))
-    c.bake(extra_context={'partition': 'normal',
-                          'output': 'logs/slurm-%j.out',
-                          'error': 'logs/slurm-%j.err'})
+    c.bake(extra_context={'sbatch_defaults': defaults})
     # Advanced setting
     c = Cookies(template, output_factory, _cookiecutter_config_file)
     c._new_output_dir = lambda: str(p.join("slurm-advanced"))
-    c.bake(extra_context={'partition': 'normal',
-                          'output': 'logs/slurm-%j.out',
-                          'error': 'logs/slurm-%j.err',
-                          'submit_script': 'slurm-submit-advanced.py'})
+    c.bake(extra_context={'sbatch_defaults': defaults,
+                          'adjust_to_partition': "True"})
```

Comment:
The constraint is currently set in cluster-config.yaml so an additional extra content is needed here, namely `'default_cluster_config': str(cluster_config),`. The test will still fail as the `--constraint` flag is not, AFAICT, passed to sbatch. This is of minor importance however; feel free to modify if you wish, but the tests need to be adjusted anyway so I can look at it later on.

Reply:
Yes true, any option parsed in the config file is parsed to sbatch. The adjust_to_partition option only updates (ie. doesnt remove) arguments. Might be something that could be changed if needed.